.github/workflows/r_package.yml

```diff
@@ -165,7 +165,7 @@ jobs:
       - name: Install packages
         shell: bash
         run: |
-          Rscript -e "install.packages(c('R6', 'data.table', 'jsonlite', 'testthat'), repos = 'https://cran.r-project.org')"
+          Rscript -e "install.packages(c('R6', 'data.table', 'jsonlite', 'testthat', 'knitr', 'rmarkdown'), repos = 'https://cran.r-project.org')"
```

Comment:
this line only controls the dependencies for a CI job using a version of R augmented with the ASAN and UBSAN sanitizers.

To make `{kintr}` and `{rmarkdown}` available to all of the other R package jobs defined in this config, can you please add these dependencies to https://github.com/microsoft/LightGBM/blob/84c4b7504a49eed63fa819a92db7d0b18292f3f4/.ci/test_r_package.sh#L102?

We hard-code the list of dependencies in these scripts instead of doing something like `remotes::install_deps()` because  the last time I checked, that function (and `devtools::install_deps()` before it) passes the option `dependencies` all the way through. So, for example, `remotes::install_deps(dependencies = c("Imports", "Depends", "LinkingTo", "Suggests"))` would install not only all of the `Suggests` of `{lightgbm}`, but also all of the `Suggests` of its dependencies, and so on and so forth. That gets to a very large number of packages pretty quickly, increasing our build times and the risk of failure (especially on Linux, where we have to build packages from source).

Reply:
okay, I added them (and push in a couple of minutes)

Reply:
oh sorry! I forgot to mention, you also need to make that change in the corresponding powershell script for Windows jobs: https://github.com/microsoft/LightGBM/blob/84c4b7504a49eed63fa819a92db7d0b18292f3f4/.ci/test_r_package_windows.ps1#L99